lib/wraft_doc/search_engine/typesense_index.ex

```diff
@@ -0,0 +1,84 @@
+defprotocol WraftDoc.SearchEngine.TypesenseIndex do
```

Comment:
```suggestion
defprotocol WraftDoc.SearchEngine.Encoder do
```

Update the filename accordingly